test/e2e/tests/settings/4byte-directory.spec.js

```diff
@@ -83,28 +78,23 @@ describe('4byte setting', function () {
         await openDapp(driver, contractAddress);
 
         // wait for deployed contract, calls and confirms a contract method where ETH is sent
-        await driver.findClickableElement('#depositButton');
         await driver.clickElement('#depositButton');
 
-        await driver.waitForSelector({
-          css: 'span',
-          text: 'Deposit initiated',
-        });
-
-        await driver.waitUntilXWindowHandles(3);
         await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
         const contractInteraction = 'Contract interaction';
         const actionElement = await driver.waitForSelector({
           css: '.confirm-page-container-summary__action__name',
           text: contractInteraction,
         });
-        // We add a delay here to wait for any potential UI changes
-        await driver.delay(veryLargeDelayMs);
-        // css text-transform: uppercase is applied to the text
-        assert.equal(
-          await actionElement.getText(),
-          contractInteraction.toUpperCase(),
-        );
```

Comment:
this way of asserting can cause race conditions, when the element it's present but not the value we expected. This is an anti-pattern, that needs to be switch by waiting for a value (text) we want. This way we can safely remove the delay